src/link/MachO.zig

```diff
@@ -660,11 +652,16 @@ pub fn resolveLibSystem(
             "libSystem",
         )) break :success;
 
-        const dir = try comp.zig_lib_directory.join(tmp_arena, &[_][]const u8{ "libc", "darwin" });
-        const lib_name = try std.fmt.allocPrint(tmp_arena, "libSystem.{d}", .{
-            self.base.options.target.os.version_range.semver.min.major,
-        });
-        if (try accessLibPath(tmp_arena, &test_path, &checked_paths, dir, lib_name)) break :success;
+        switch (self.base.options.darwin_sdk_layout.?) {
```

Comment:
Is it possible to reach this point with `darwin_sdk_layout == null`? If so, we should try unwrapping without panicking and let the linker report the error via `reportMissingLibraryError`. In general, we should only ever panic if something is never allowed to happen and was an oversight such as lack of `SDKSettings.json` for vendored SDK which you have already added.